tests/e2e/testsLibrary/WorkspaceHandlingTests.ts

```diff
@@ -15,46 +15,109 @@ import { Dashboard } from '../pageobjects/dashboard/Dashboard';
         @inject(CLASSES.CreateWorkspace) private readonly createWorkspace: CreateWorkspace,
         @inject(CLASSES.Workspaces) private readonly workspaces: Workspaces,
-        @inject(CLASSES.BrowserTabsUtil) private readonly browserTabsUtil: BrowserTabsUtil) {}
+        @inject(CLASSES.BrowserTabsUtil) private readonly browserTabsUtil: BrowserTabsUtil,
+        @inject(CLASSES.ApiUrlResolver) private readonly apiUrlResolver: ApiUrlResolver,
+        @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper,
+        @inject(CLASSES.Ide) private readonly ide: Ide) {}
 
     public createAndOpenWorkspace(stack: string) {
-        test(`Open 'New Workspace' page`, async () => {
+        test(`Create and open new workspace, stack:${stack}`, async () => {
             await this.dashboard.waitPage();
+            Logger.debug(`Fetching user kubernetes namespace, storing auth token by getting workspaces API URL.`);
+            await this.apiUrlResolver.getWorkspacesApiUrl();
             await this.dashboard.clickCreateWorkspaceButton();
             await this.createWorkspace.waitPage();
-            const parentGUID = await this.browserTabsUtil.getCurrentWindowHandle();
+            WorkspaceHandlingTests.parentGUID = await this.browserTabsUtil.getCurrentWindowHandle();
             await this.createWorkspace.clickOnSample(stack);
-            await this.browserTabsUtil.switchToWindow(parentGUID);
-            WorkspaceHandlingTests.workspaceName = await this.dashboard.getRecentWorkspaceName(10000);
-            Logger.debug(`Workspace Name is: ${WorkspaceHandlingTests.workspaceName}`);
-            await this.browserTabsUtil.waitAndSwitchToAnotherWindow(parentGUID, 10000);
+            await this.browserTabsUtil.waitAndSwitchToAnotherWindow(WorkspaceHandlingTests.parentGUID, TimeoutConstants.TS_IDE_LOAD_TIMEOUT);
         });
     }
 
     public openExistingWorkspace(workspaceName: string) {
-        test('Start workspace', async () => {
+        test('Open and start existing workspace', async () => {
             await this.dashboard.waitPage();
+            Logger.debug(`Fetching user kubernetes namespace, storing auth token by getting workspaces API URL.`);
+            await this.apiUrlResolver.getWorkspacesApiUrl();
             await this.dashboard.clickWorkspacesButton();
             await this.workspaces.waitPage();
             await this.workspaces.clickOpenButton(workspaceName);
         });
     }
 
+    public obtainWorkspaceNameFromStartingPage() {
+        test('Obtain workspace name from workspace loader page', async() => {
+            try {
+                await this.driverHelper.waitVisibility(WorkspaceHandlingTests.START_WORKSPACE_PAGE_NAME_LOCATOR, TimeoutConstants.TS_WAIT_LOADER_PRESENCE_TIMEOUT);
+                // it takes a while to update the element with the workspace name
+                await this.driverHelper.wait(10_000);
```

Comment:
Can `10_000` be replaced with existed timeout constant like `TS_IDE_LOAD_TIMEOUT`, not to deal with "magic number"?

Reply:
done

Reply:
thanks